core/src/Cuda/Kokkos_Cuda_ReduceScan.hpp

```diff
@@ -501,7 +501,7 @@ struct CudaReductionsFunctor<FunctorType, ArgTag, false, true> {
           1;
     }
     bool is_last_block = false;
-    if (__syncthreads_or(num_teams_done == gridDim.x)) {
+    if (__syncthreads_or(num_teams_done == gridDim.x ? 1 : 0) != 0) {
```

Comment:
if you must cast to `int`
```suggestion
    if (__syncthreads_or(static_cast<int>(num_teams_done == gridDim.x)) != 0) {
```

Reply:
I don't care about that too much (even if I prefer the current version for clarity).